components/analytics-solutions-common/org.wso2.analytics.solutions.common/widgets/DateTimeRangePicker/src/DateRangePicker.jsx

```diff
@@ -211,100 +244,91 @@ export default class DateRangePicker extends Widget {
     }
 
     loadDefaultTimeRange() {
-        if (location.search !== '') {
-            let dateTimeRangeInfo = JSON.parse(decodeURI(location.search.substr(1)));
-            if (dateTimeRangeInfo.hasOwnProperty('tr')) {
-                if (dateTimeRangeInfo.tr.toLowerCase() === 'custom') {
-                    if (dateTimeRangeInfo.hasOwnProperty('sd')
-                        && dateTimeRangeInfo.hasOwnProperty('ed')) {
-                        if (dateTimeRangeInfo.hasOwnProperty('g')) {
-                            this.loadUserSpecifiedCustomTimeRange(dateTimeRangeInfo.sd,
-                                dateTimeRangeInfo.ed, dateTimeRangeInfo.g)
-                        } else {
-                            this.loadUserSpecifiedCustomTimeRange(dateTimeRangeInfo.sd,
-                                dateTimeRangeInfo.ed, '')
-                        }
+        const dateTimeRangeInfo = this.getDateTimeRangeInfo();
+        if (dateTimeRangeInfo.tr) {
+            if (dateTimeRangeInfo.tr.toLowerCase() === 'custom') {
+                if (dateTimeRangeInfo.sd
+                    && dateTimeRangeInfo.ed) {
+                    if (dateTimeRangeInfo.g) {
+                        this.loadUserSpecifiedCustomTimeRange(dateTimeRangeInfo.sd,
+                            dateTimeRangeInfo.ed, dateTimeRangeInfo.g);
```

Comment:
Can't we simplify this logic?

Reply:
This can't be further simplified as the 'else' for each if statement behaves differently